model-optimizer/mo/ops/slice.py

```diff
@@ -13,133 +13,181 @@
  See the License for the specific language governing permissions and
  limitations under the License.
 """
-
-import logging as log
+from typing import List
 
 import numpy as np
 
 from mo.graph.graph import Node, Graph
 from mo.ops.op import Op
+from mo.utils.error import Error
+
+"""
+Slicing operations have different semantic or different parameters/inputs in different frameworks. To distinguish them 
+several internal operations are introduced. The internal MO Slice operation behaves same as Slice in ONNX opset >= 10. 
+A number of transformations take place on the front phase to convert framework slicing:
+ - AttributedSlice, TFSlice -> Slice 
+ - CaffeSlice -> Split 
+ - MXSlice -> StridedSlice 
+"""
+
+
+class AttributedSlice(Op):
+    """
+    AttributedSlice is used in old versions of ONNX models (opset version < 10).
+    The operation is replaced with the OpenVINO Slice operation on the front phase.
+    """
+    op = 'AttributedSlice'
+    enabled = False
+
+    def __init__(self, graph: Graph, attrs: dict):
+        super().__init__(graph, {
+            'type': None,
+            'op': self.op,
+            'in_ports_count': 1,
+            'out_ports_count': 1,
+            'infer': None,
+        }, attrs)
+
+    def supported_attrs(self):
```

Comment:
We won't emit this op to IR, so it is useless. Should we remove this method? For making sure all of the attributes are set you may have an assertion in __init__ of the class.

Reply:
Ok, thanks for noting this. Removed. Did the same for the CaffeSlice